apps/back-end/src/services/Dataset.ts

```diff
@@ -90,12 +89,7 @@ export class Dataset {
 
   getItem = (itemIx: number) => {
     if (!fs.existsSync(path.join(this.folderPath, "items", `${itemIx}.json`))) {
-      throw new TsRestResponseError(contract.getDatasetItem, {
-        status: 404,
-        body: {
-          message: `can't retrieve data for dataset ${this.id} item @${itemIx}`,
-        },
-      });
+      throw new HttpError(404, `Can't retrieve data for dataset item`);
```

Comment:
I suggest adding the ID back in, otherwise it makes it a bit harder to see if the ID isn't being interpreted correctly. (And at your suggestion, I'll add this in a bit.)